logstash-core/lib/logstash/output_delegator.rb

```diff
@@ -12,7 +12,8 @@ def initialize(logger, output_class, metric, strategy_registry, plugin_args)
     @output_class = output_class
     @metric = metric
     @plugin_args = plugin_args
-    @strategy_registry ||= ::LogStash::OutputDelegatorStrategyRegistry.instance
+    @strategy_registry = strategy_registry
```

Comment:
I am cautious of this variable name, btw. It shadows the accessor method `strategy_registry`. I don't have any particular recommendations, though. :\